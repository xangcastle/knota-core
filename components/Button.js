import React from "react";

const Button = function ({onClick, text, style}) {
    return <div className={'button'}
                testID={'button-div'}
                style={{
                    backgroundColor: 'red',
                    padding: '20px',
                    borderRadius: '17px',
                    color: 'white',
                    fontWeight: 'bold',
                    textTransform: 'uppercase',
                    border: '2px solid beige',
                    cursor: 'pointer', ...style
                }} onClick={onClick}>{text}</div>
}

export default Button;
