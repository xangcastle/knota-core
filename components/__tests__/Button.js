import React from "react";
import Button from "../Button";
import TestRenderer from 'react-test-renderer';

describe("Testing button component", ()=>{
    let wrapper;
    let onClick;

    beforeEach(()=>{
        onClick = jest.fn();
        wrapper = TestRenderer.create(<Button text={'My button'} onClick={onClick} />);
    });

    it("Button component should render successfully", () =>{
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it("Button component should handle on click event", () =>{
        TestRenderer.act(()=>{
            const button = wrapper.root.findByProps({
                testID: "button-div"
            });
            button.props.onClick();
            expect(onClick).toBeCalled();
        });
    });
})
