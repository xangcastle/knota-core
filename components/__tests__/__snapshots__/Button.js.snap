// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing button component Button component should render successfully 1`] = `
<div
  className="button"
  onClick={[MockFunction]}
  style={
    Object {
      "backgroundColor": "red",
      "border": "2px solid beige",
      "borderRadius": "17px",
      "color": "white",
      "cursor": "pointer",
      "fontWeight": "bold",
      "padding": "20px",
      "textTransform": "uppercase",
    }
  }
  testID="button-div"
>
  My button
</div>
`;
